tools.py

```diff
@@ -6,13 +7,17 @@
 import itertools
 import subprocess
 import collections
+from functools import partial
 import torch
 import sentencepiece
+import tqdm
```

Comment:
If you start using tqdm, please put it into Dockerfile as a dependency (IIRC it's not there now)

Reply:
fixed